src/loaders/genome_collection/parse_biolog_data.py

```diff
@@ -0,0 +1,187 @@
+    COLL_MICROTRAIT_DATA,
+    COLL_MICROTRAIT_META,
+    FLD_ARANGO_KEY,
+    FLD_COLLECTION_ID,
+    FLD_LOAD_VERSION,
+)
+from src.common.storage.db_doc_conversions import collection_load_version_key
+from src.common.storage.field_names import FLD_KBASE_ID
+from src.loaders.common import loader_common_names
+from src.loaders.common.loader_helper import init_row_doc, create_import_files
+from src.loaders.genome_collection.parse_tool_results import HEATMAP_FILE_ROOT
+
+
+def _read_excel_as_df(excel_file: Path, sheet_name=0) -> pd.DataFrame:
+    # Read the Excel file into a DataFrame
+
+    try:
+        df = pd.read_excel(excel_file, sheet_name=sheet_name)
+
+        return df
+
+    except FileNotFoundError:
+        print(f"File '{excel_file}' not found.")
+
+
+def _read_biolog_data(biolog_data_file: Path) -> pd.DataFrame:
+    # Read the Biolog data file into a DataFrame and do some data cleaning
+
+    data_df = _read_excel_as_df(biolog_data_file)
+
+    # NOTE: below logic is specific to the Biolog data file (PMI strain BiologSummary.xlsx)
+    # link to the file: https://docs.google.com/spreadsheets/d/1QmC6UHWOEVfpmrveRBl_izictbNeN1PA/edit#gid=1135979967
+
+    # Drop unused columns
+    data_df = data_df.drop(columns=['# pos. strains'])
+
+    # Make the first row ('strain designation') the column names
+    data_df.columns = data_df.loc[1]
+
+    # Drop unused rows ('total cpds. Utilized', 'strain designation' and unused rows at the end.)
+    data_df = data_df.drop(index=[0, 1, 194, 195, 196, 197, 198])
+
+    # Make the first column ('strain designation') the index
+    data_df = data_df.set_index('strain designation')
+
+    # Transpose the DataFrame
+    data_df.index.name = None
+    data_df.columns.name = None
+    data_df = data_df.transpose()
+
+    # Remove asterisks from index names
+    data_df.index = data_df.index.str.replace(r'*', '').str.replace(r'**', '')
+
+    # Merge rows BT03 A(10/1) and BT03 B(11/18) into one row BT03
+    data_df = data_df.drop(index='BT03 B(11/18)')
+    data_df = data_df.rename(index={'BT03 A(10/1)': 'BT03'})
+
+    # Replace NaN with 0
+    # TODO: this is a temporary solution to handle the missing data for some strains.
+    # Two NaN values are present in the data file for 'YR530' and 'CF286' in the row labeled '3-Hydroxy 2-Butanone'.
+    data_df.fillna(0, inplace=True)
+
+    return data_df
+
+
+def _read_biolog_meta(biolog_meta_file: Path) -> pd.DataFrame:
+    meta_df = _read_excel_as_df(biolog_meta_file)
+
+    return meta_df
+
+
+def _retrieve_kbase_genome_id(meta_df, strain_id) -> str:
+    # Retrieve the KBase genome ID for the given strain ID from the metadata file (dataframe).
+
+    matching_row = meta_df[(meta_df['Strain ID'] == strain_id) & meta_df['Genome_Ref'].notna()]
+
+    if matching_row.empty:
+        raise ValueError(f"No matching Strain ID with a non-NaN Genome_Ref found for {strain_id}.")
+    else:
+        genome_ref = matching_row['Genome_Ref'].values[0]
```

Comment:
Aren't we using the assembly ref as the kbase ID everywhere else?

Reply:
updated.